cpp/include/cudf/strings/split/split.hpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+
+namespace cudf
+{
+namespace strings
+{
+
+/**
+ * @brief Returns a list of columns by splitting each string using the
+ * specified delimiter.
+ *
+ * The number of rows in the output columns will be the same as the
+ * input column. The first column will contain the first tokens of
+ * each string as a result of the split. Subsequent columns contain
+ * the next token strings. Null entries are added for a row where
+ * split results have been exhausted. The total number of columns
+ * will equal the maximum number of splits encountered on any string
+ * in the input column.
+ *
+ * Any null string entries return corresponding null output columns.
+ *
+ * @param strings Strings instance for this operation.
+ * @param delimiter UTF-8 encoded string indentifying the split points in each string.
+ *        Default of empty string indicates split on whitespace.
+ * @param maxsplit Maximum number of splits to perform.
+ *        Default of -1 indicates all possible splits on each string.
+ * @param mr Resource for allocating device memory.
+ * @return New strings columns.
+ */
+std::vector<std::unique_ptr<column>> split( strings_column_view const& strings,
```

Comment:
Will it be helpful if we return table rather than vector of columns?

Reply:
I was just thinking about that today. All the columns should be the same size so I think returning a table would be right thing here. I have a couple of other PRs that should do the same as well.